api/v1alpha1/hostedcluster_types.go

```diff
@@ -38,9 +38,11 @@ type HostedClusterSpec struct {
 }
 
 type ClusterNetworking struct {
-	ServiceCIDR string `json:"serviceCIDR"`
-	PodCIDR     string `json:"podCIDR"`
-	MachineCIDR string `json:"machineCIDR"`
+	ServiceCIDR                string `json:"serviceCIDR"`
+	PodCIDR                    string `json:"podCIDR"`
+	MachineCIDR                string `json:"machineCIDR"`
+	APIServerAdvertisedAddress string `json:"apiserverAdvertisedAddress,omitempty"`
+	APIServerSecurePort        uint   `json:"apiserverSecurePort,omitempty"`
```

Comment:
@csrwng @enxebre @sjenning @derekwaynecarr Just wanted to quickly note that I think we need to focus on this API change and make sure we are thinking through the model. I'm not sure we've discussed this specifically yet. I think I need some more context.

Reply:
They can vary on a cluster per cluster basis. Currently in ROKS however they are usually set to 2040 and 172.20.0.1.